File: sample/javaSample/src/main/java/jp/co/geniee/samples/gnad/interstitial/InterstitialDemoActivity.java
package jp.co.geniee.samples.gnad.interstitial;

import android.content.Context;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import jp.co.geniee.samples.R;
import jp.co.geniee.samples.SharedPreferenceManager;
import jp.co.geniee.gnadsdk.common.GNAdLogger;
import jp.co.geniee.gnadsdk.interstitial.GNInterstitial;

public class InterstitialDemoActivity extends AppCompatActivity implements GNInterstitial.GNInterstitialListener, GNInterstitial.GNInterstitialDialogListener, View.OnClickListener {

    private final String TAG = "[GNS]InterstitialDemo";

    private Context mContext;

    private EditText edtZoneId;
    private Button btLoadGNAd, btShowGNAd;

    private GNInterstitial mGnInterstitial;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_gnad_interstitial);

        mContext = this;

        edtZoneId = findViewById(R.id.edtZoneId);
        edtZoneId.setText(SharedPreferenceManager.getInstance(mContext).getString(SharedPreferenceManager.INTERSTITIAL_AD_ZONE_ID));
        btLoadGNAd = findViewById(R.id.btLoadGNAd);
        btShowGNAd = findViewById(R.id.btShowGNAd);

        btLoadGNAd.setOnClickListener(this);
        btShowGNAd.setOnClickListener(this);
    }

    @Override
    public void onReceiveSetting() {
        btShowGNAd.setEnabled(true);

        Log.d(TAG, "onReceiveSetting");

        Toast.makeText(mContext, "Interstitial ad received", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onFailedToReceiveSetting() {
        btShowGNAd.setEnabled(false);

        Log.d(TAG, "onFailedToReceiveSetting");

        Toast.makeText(mContext, "Interstitial ad failed to receive", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onButtonClick(int i) {

        Log.d(TAG, "onButtonClick");
    }

    @Override
    public void onClose() {

        Log.d(TAG, "onClose interstitial");
        btShowGNAd.setEnabled(false);
    }

    @Override
    public void onClick(View view) {

        if (view.getId() == R.id.btLoadGNAd) {

            prepareInterstitialAd();

        } else if (view.getId() == R.id.btShowGNAd) {

            if (mGnInterstitial.isReady()) {
                mGnInterstitial.show(this);
            }
        }

    }

    private void prepareInterstitialAd() {

        try {
            mGnInterstitial = new GNInterstitial(this, Integer.parseInt(edtZoneId.getText().toString()));
            mGnInterstitial.setListener(this);
            mGnInterstitial.setDialoglistener(this);
            mGnInterstitial.load(this);
            mGnInterstitial.setGeoLocationEnable(true);
            mGnInterstitial.setLogPriority(GNAdLogger.INFO);

            SharedPreferenceManager.getInstance(mContext).putString(SharedPreferenceManager.INTERSTITIAL_AD_ZONE_ID, edtZoneId.getText().toString());
        } catch (Exception e) {
            edtZoneId.setError(e.getLocalizedMessage());
        }

    }
}

File: sample/kotlinSample/src/main/java/jp/co/geniee/samples/gnad/interstitial/InterstitialDemoActivity.kt
package jp.co.geniee.samples.gnad.interstitial

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import jp.co.geniee.samples.R
import jp.co.geniee.samples.SharedPreferenceManager
import jp.co.geniee.gnadsdk.common.GNAdLogger
import jp.co.geniee.gnadsdk.interstitial.GNInterstitial

class InterstitialDemoActivity : AppCompatActivity(), GNInterstitial.GNInterstitialListener, GNInterstitial.GNInterstitialDialogListener, View.OnClickListener {

    private val TAG = "[GNS]InterstitialDemo"

    private lateinit var mContext: Context

    private lateinit var mGnInterstitial: GNInterstitial

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_interstitial_demo)

        mContext = this

        val edtZoneId = findViewById<TextView>(R.id.edtZoneId)
        edtZoneId.setText(SharedPreferenceManager.getInstance(mContext).getString(SharedPreferenceManager.INTERSTITIAL_AD_ZONE_ID))

        val btLoadGNAd = findViewById<Button>(R.id.btLoadGNAd)
        btLoadGNAd.setOnClickListener(this)
        val btShowGNAd = findViewById<Button>(R.id.btShowGNAd)
        btShowGNAd.setOnClickListener(this)
    }

    override fun onReceiveSetting() {
        val btShowGNAd = findViewById<Button>(R.id.btShowGNAd)
        btShowGNAd.isEnabled = true

        Log.d(TAG, "onReceiveSetting")

        Toast.makeText(mContext, "Interstitial ad received", Toast.LENGTH_SHORT).show()
    }

    override fun onFailedToReceiveSetting() {
        val btShowGNAd = findViewById<Button>(R.id.btShowGNAd)
        btShowGNAd!!.isEnabled = false

        Log.d(TAG, "onFailedToReceiveSetting")

        Toast.makeText(mContext, "Interstitial ad failed to receive", Toast.LENGTH_SHORT).show()
    }

    override fun onButtonClick(i: Int) {

        Log.d(TAG, "onButtonClick")
    }

    override fun onClose() {

        Log.d(TAG, "onClose interstitial")
        val btShowGNAd = findViewById<Button>(R.id.btShowGNAd)
        btShowGNAd!!.isEnabled = false
    }

    override fun onClick(view: View) {

        if (view.id == R.id.btLoadGNAd) {

            prepareInterstitialAd()

        } else if (view.id == R.id.btShowGNAd) {

            if (mGnInterstitial.isReady) {
                mGnInterstitial.show(this)
            }
        }

    }

    private fun prepareInterstitialAd() {
        val edtZoneId = findViewById<TextView>(R.id.edtZoneId)
        try {
            mGnInterstitial = GNInterstitial(this, Integer.parseInt(edtZoneId.text.toString()))
            mGnInterstitial.listener = this
            mGnInterstitial.dialoglistener = this
            mGnInterstitial.load(this)
            mGnInterstitial.setGeoLocationEnable(true)
            mGnInterstitial.setLogPriority(GNAdLogger.INFO)

            SharedPreferenceManager.getInstance(mContext).putString(SharedPreferenceManager.INTERSTITIAL_AD_ZONE_ID, edtZoneId.text.toString())
        } catch (e: Exception) {
            edtZoneId.error = e.localizedMessage
        }

    }
}
